urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/db/DBConnection.java

```diff
@@ -0,0 +1,584 @@
+         *  PARAM: [IN] object - serializable object
+         * AUTHOR: Dmitry Eliseev
+         * */
+        public final boolean setObject(final int queryIndex, final int index, final Serializable object) {
+            final byte[] serializedObject = CompressionUtility.compress(object);
+            return serializedObject != null && setBinaryStream(queryIndex, index, new ByteArrayInputStream(serializedObject));
+        } /* End of 'DBMultipleStatement::setObject' method */
+
+        /* *
+         * METHOD: Returns the number of statements in batch
+         * RETURN: The number of statements in batch
+         * AUTHOR: Dmitry Eliseev
+         * */
+        public final int size() {
+            return m_Statements.length;
+        } /* End of 'DBMultipleStatement::size' method */
+    } /* End of 'DBConnection::DBMultipleStatement' class */
+
+    /* *
+     * METHOD: Class constructor
+     *  PARAM: [IN] host - Database service host
+     *  PARAM: [IN] port - Database service port
+     *  PARAM: [IN] name - Database name
+     *  PARAM: [IN] user - Database user's name
+     *  PARAM: [IN] pass - Database user's password
+     * AUTHOR: Dmitry Eliseev
+     * */
+    public DBConnection(final String host, final int port, final String name, final String user, final String pass) {
+        final BoneCPConfig config = new BoneCPConfig();
+        config.setJdbcUrl("jdbc:mysql://" + host + ":" + port + "/" + name + "?useUnicode=yes&characterEncoding=UTF-8&useOldAliasMetadataBehavior=true");
+        config.setUsername(user);
+        config.setPassword(pass);
+        config.setInitSQL("SET CHARACTER SET utf8, character_set_client = utf8, character_set_connection = utf8, " +
+                          "character_set_results = utf8, NAMES utf8, collation_connection = utf8_general_ci, " +
+                          "collation_database = utf8_general_ci");
+
+        /* Pool size configuration */
+        config.setMaxConnectionsPerPartition(5);
+        config.setMinConnectionsPerPartition(5);
+        config.setPartitionCount(1);
+
+        try {
+            m_Pool = new BoneCP(config);
+        } catch (final SQLException e) {
+            LoggingUtility.error("Can't initialize connections pool: " + e.getMessage());
+            m_Pool = null;
+        }
+    } /* End of 'DBConnection::DBConnection' method */
+
+    @Override
+    protected final void finalize() throws Throwable {
+        super.finalize();
+
+        if (m_Pool != null)
+            m_Pool.shutdown();
+    } /* End of 'DBConnection::finalize' method  */
+
+    /* *
+     * METHOD: Prepares statement using current connection
+     * RETURN: Prepared statement
+     *  PARAM: [IN] query - SQL query
+     * AUTHOR: Dmitry Eliseev
+     * */
+    public final DBStatement createStatement(final String query) {
+        if (ValidationUtility.isEmpty(query))
+            return null;
+
+        try {
+            final Connection connection = m_Pool.getConnection();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *connection*. Currently, there are execution paths that do not contain closure statements, for example, when Connection.prepareStatement throws an exception. Ensure *connection* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).